Support for COMP_POINT seems to have broken completion…

```panic: runtime error: slice bounds out of range

goroutine 1 [running]:
.../vendor/github.com/posener/complete.(*Complete).Complete(0xc00038c690, 0x0)
	.../vendor/github.com/posener/complete/complete.go:67 +0x634
...
panic: runtime error: slice bounds out of range```

on tabbing for completion right after the name of the command. I haven't investigated closer but reverting to `1.1.2` fixes the issue.

I use zsh:
> zsh --version
zsh 5.3 (x86_64-apple-darwin18.0)

The code doesn't get to completion so `COMP_DEBUG` doesn't do anything, the error is the same as the one I've been getting (typing `./self` space then tab):

```
./self panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/posener/complete.(*Complete).Complete(0xc0000b0000, 0x0)
	/Users/claprun/Dev/go/src/github.com/posener/complete/complete.go:67 +0x634
main.main()
	/Users/claprun/Dev/go/src/github.com/posener/complete/example/self/main.go:42 +0x23f
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/posener/complete.(*Complete).Complete(0xc0000be000, 0x0)
	/Users/claprun/Dev/go/src/github.com/posener/complete/complete.go:67 +0x634
main.main()
	/Users/claprun/Dev/go/src/github.com/posener/complete/example/self/main.go:42 +0x23f
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/posener/complete.(*Complete).Complete(0xc000098000, 0x0)
	/Users/claprun/Dev/go/src/github.com/posener/complete/complete.go:67 +0x634
main.main()
	/Users/claprun/Dev/go/src/github.com/posener/complete/example/self/main.go:42 +0x23f```

--- a/complete.go
+++ b/complete.go
@@ -64,10 +64,12 @@ func (c *Complete) Complete() bool {
 		return c.CLI.Run()
 	}
 
-	completePhrase := line[:point]
+	if point >= 0 && point < len(line) {
+		line = line[:point]
+	}
 
-	Log("Completing phrase: %s", completePhrase)
-	a := newArgs(completePhrase)
+	Log("Completing phrase: %s", line)
+	a := newArgs(line)
 	Log("Completing last field: %s", a.Last)
 	options := c.Command.Predict(a)
 	Log("Options: %s", options)

